박은지 week2 과제 제출합니다.

eunjipark123 opened the pull request.

eunjipark123 committed:
과제 작성

SQL 쿼리 작성

Moon-edu commented:
오답 처리된 게 몇 개 있습니다. 에러 메시지를 보고 수정해주시면 되겠습니다.
3번
```
2023-11-22 09:35:15 ERROR column "emp_id" does not exist
LINE 8:       visitor char(6) check(emp_id ~ '^[A-Z][0-9]{5}+$'),
```

Moon-edu commented on homework/q1_create_employee_table.sql:
```diff
@@ -9,3 +9,15 @@
 
+
+create table employee (
+    emp_id char(6) unique not null check(emp_id ~ '^[A-Z][0-9]{5}+$'),
```
`+`가 왜 있는지 모르겠습니다. 
`^[A-Z][0-9]{5}+$`여야 할 것 같습니다.
^[A-Z] : 대문자로 시작
[0-9]{5} 숫자 다섯자리
$ 끝
---

Moon-edu replied:
`+`가 왜 있는지 모르겠습니다.
`^[A-Z][0-9]{5}$`여야 할 것 같습니다.
`^[A-Z]` : 대문자로 시작
`[0-9]{5}` 숫자 다섯자리
`$` 끝

Moon-edu commented on homework/q1_create_employee_table.sql:
```diff
@@ -9,3 +9,15 @@
+    name varchar(20) not null check(name ~ '^[A-Za-z]+$'),
+    address varchar(100) check(address ~ '^[\w\s,-/]+$'),
+    department int check(department < 100),
+    manager char(6) check(emp_id ~ '^[A-Z][0-9]{5}+$'),
```
같은 이유로 잘못된 regex입니다

eunjipark123 committed:
과제 수정

eunjipark123 committed:
과제 수정 제출

eunjipark123 commented:
다시 제출합니다

Moon-edu commented:
몇 가지 요건을 충족을 시키지 못해 오답 처리된 항목 알려드립니다. 에러로그 보시고 어떤게 잘못되었는지 디버깅해보세요
1. Q3
```
2023-11-23 15:14:02 ERROR new row for relation "employee" violates check constraint "employee_address_check"
DETAIL:  Failing row contains (A00001, Male, Moon, 10-199, Gang-nam, Seoul, 1, C00001, 30, Senior engineer).
FAILED       
```

Moon-edu commented:
Q4
```
2023-11-23 15:14:02 WARNING Insert 후 결과 값이 예상 값이 아닙니다. Expected [(None, datetime.datetime(2022, 7, 11, 12, 30), datetime.datetime(2022, 7, 11, 14, 45), 'meeting'), (None, datetime.datetime(2022, 7, 12, 15, 20), datetime.datetime(2022, 7, 12, 16, 50), 'visit family'), ('A00001', datetime.datetime(2022, 7, 11, 11, 0), datetime.datetime(2022, 7, 11, 18, 0), 'work'), ('A00001', datetime.datetime(2022, 7, 12, 8, 15), datetime.datetime(2022, 7, 12, 17, 56), 'work'), ('A00001', datetime.datetime(2022, 7, 13, 9, 0), datetime.datetime(2022, 7, 13, 18, 20), 'work'), ('A00001', datetime.datetime(2022, 7, 14, 8, 30), None, 'work'), ('A08771', datetime.datetime(2022, 7, 11, 9, 0), datetime.datetime(2022, 7, 11, 17, 56), 'work'), ('A08771', datetime.datetime(2022, 7, 12, 9, 20), datetime.datetime(2022, 7, 12, 18, 20), 'work'), ('A08771', datetime.datetime(2022, 7, 13, 9, 20), datetime.datetime(2022, 7, 13, 18, 0), 'work'), ('A08771', datetime.datetime(2022, 7, 14, 8, 30), None, 'work'), ('B00100', datetime.datetime(2022, 7, 11, 10, 0), datetime.datetime(2022, 7, 11, 19, 0), 'work'), ('B00100', datetime.datetime(2022, 7, 12, 8, 30), datetime.datetime(2022, 7, 12, 19, 0), 'work'), ('B00100', datetime.datetime(2022, 7, 13, 8, 30), datetime.datetime(2022, 7, 13, 18, 0), 'work'), ('B00100', datetime.datetime(2022, 7, 14, 8, 30), None, 'work'), ('C00001', datetime.datetime(2022, 7, 11, 9, 20), datetime.datetime(2022, 7, 11, 18, 0), 'work'), ('C00001', datetime.datetime(2022, 7, 12, 10, 0), datetime.datetime(2022, 7, 12, 18, 0), 'work'), ('C00001', datetime.datetime(2022, 7, 13, 9, 20), datetime.datetime(2022, 7, 13, 18, 20), 'work'), ('C00002', datetime.datetime(2022, 7, 11, 9, 30), datetime.datetime(2022, 7, 11, 18, 20), 'work'), ('C00002', datetime.datetime(2022, 7, 12, 9, 0), datetime.datetime(2022, 7, 12, 17, 56), 'work'), ('C00002', datetime.datetime(2022, 7, 13, 8, 15), datetime.datetime(2022, 7, 13, 17, 56), 'work'), ('C00129', datetime.datetime(2022, 7, 11, 9, 0), datetime.datetime(2022, 7, 11, 18, 20), 'work'), ('C00129', datetime.datetime(2022, 7, 12, 10, 0), datetime.datetime(2022, 7, 12, 18, 0), 'work'), ('C00129', datetime.datetime(2022, 7, 13, 10, 0), datetime.datetime(2022, 7, 13, 18, 0), 'work')], Actual [(None, datetime.datetime(2022, 7, 11, 12, 30), datetime.datetime(2022, 7, 11, 14, 45), 'meeting'), (None, datetime.datetime(2022, 7, 12, 15, 20), datetime.datetime(2022, 7, 12, 16, 50), 'visit family'), ('A00001', datetime.datetime(2022, 7, 11, 11, 0), datetime.datetime(2022, 7, 11, 18, 0), 'work'), ('A00001', datetime.datetime(2022, 7, 12, 9, 20), datetime.datetime(2022, 7, 12, 19, 0), 'work'), ('A00001', datetime.datetime(2022, 7, 13, 9, 0), datetime.datetime(2022, 7, 13, 18, 20), 'work'), ('A00001', datetime.datetime(2022, 7, 14, 8, 30), None, 'work'), ('A08771', datetime.datetime(2022, 7, 11, 9, 0), datetime.datetime(2022, 7, 11, 17, 56), 'work'), ('A08771', datetime.datetime(2022, 7, 12, 9, 20), datetime.datetime(2022, 7, 12, 18, 20), 'work'), ('A08771', datetime.datetime(2022, 7, 13, 9, 20), datetime.datetime(2022, 7, 13, 18, 0), 'work'), ('A08771', datetime.datetime(2022, 7, 14, 8, 30), None, 'work'), ('B00100', datetime.datetime(2022, 7, 11, 10, 0), datetime.datetime(2022, 7, 11, 19, 0), 'work'), ('B00100', datetime.datetime(2022, 7, 12, 9, 20), datetime.datetime(2022, 7, 12, 17, 56), 'work'), ('B00100', datetime.datetime(2022, 7, 13, 8, 30), datetime.datetime(2022, 7, 13, 18, 0), 'work'), ('B00100', datetime.datetime(2022, 7, 14, 8, 30), None, 'work'), ('C00001', datetime.datetime(2022, 7, 11, 9, 20), datetime.datetime(2022, 7, 11, 18, 0), 'work'), ('C00001', datetime.datetime(2022, 7, 12, 10, 0), datetime.datetime(2022, 7, 12, 18, 0), 'work'), ('C00001', datetime.datetime(2022, 7, 13, 9, 20), datetime.datetime(2022, 7, 13, 18, 20), 'work'), ('C00002', datetime.datetime(2022, 7, 11, 9, 30), datetime.datetime(2022, 7, 11, 18, 20), 'work'), ('C00002', datetime.datetime(2022, 7, 12, 9, 0), datetime.datetime(2022, 7, 12, 17, 56), 'work'), ('C00002', datetime.datetime(2022, 7, 13, 8, 15), datetime.datetime(2022, 7, 13, 17, 56), 'work'), ('C00129', datetime.datetime(2022, 7, 11, 9, 0), datetime.datetime(2022, 7, 11, 18, 20), 'work'), ('C00129', datetime.datetime(2022, 7, 12, 10, 0), datetime.datetime(2022, 7, 12, 18, 0), 'work'), ('C00129', datetime.datetime(2022, 7, 13, 10, 0), datetime.datetime(2022, 7, 13, 18, 0), 'work')]
```

Moon-edu commented on homework/q1_create_employee_table.sql:
```diff
@@ -9,3 +9,15 @@
+    manager char(6) check(emp_id ~ '^[A-Z][0-9]{5}$'),
+    age int not null check(age < 200),
+    position varchar(30)
+)
```
몇몇 Regex들에서 어떤 문자가 허용되는지는 잘 정의된 듯 한데 갯수가 틀린 것 같아요. 예를 들면 `^[A-Za-z]$`같은 경우는 알파벳 대소문자 1글자만을 뜻합니다. `1글자 이상`을 의미하는 어떤 특수문자가 [] 다음에 와야할 것 같아요 :)

eunjipark123 commented:
> 몇 가지 요건을 충족을 시키지 못해 오답 처리된 항목 알려드립니다. 에러로그 보시고 어떤게 잘못되었는지 디버깅해보세요
> 
> 1. Q3
> 
> ```
> 2023-11-23 15:14:02 ERROR new row for relation "employee" violates check constraint "employee_address_check"
> DETAIL:  Failing row contains (A00001, Male, Moon, 10-199, Gang-nam, Seoul, 1, C00001, 30, Senior engineer).
> FAILED       
> ```

이건 Q1번(employee 테이블 정의) 에 관련된 오류인 거겠죠? Q3은 visit_log에 대한 문제인데, 여기서는 이슈가 없어 보여서요!

eunjipark123 committed:
과제 수정 제출(2)

1. Q1 employee 테이블에서 name, address 정규표현식 수정했습니다. 말씀하신 것처럼 뒤에 어떤 문자가 와야하는데, +가 앞에서 나온 문자가 반복된다는 의미여서 이걸로 시도해보았습니다. Dbeaver에서는 문제가 발생하지 않았습니다.

2.Q4 인서트 항목에서 오타가 발생하여 수정했습니다.

Moon-edu commented:
전부 정답 처리되었습니다. 고생하셨어요

eunjipark123 commented:
여러 번 봐주시느라 정말 고생 많으셨습니다. 감사드립니다!
